crates/sla/src/lib.rs

```diff
@@ -421,16 +419,24 @@ impl<T: Config> Module<T> {
     /// * `amount` - the amount of polkabtc that was issued
     /// * `vault_id` - account of the vault
     fn _executed_issue_sla_change(amount: PolkaBTC<T>) -> Result<T::SignedFixedPoint, DispatchError> {
+        let amount_raw = Self::polkabtc_to_u128(amount)?;
+
         // update the number of issues performed
-        let mut count = <TotalIssueCount>::get();
-        count = count.checked_add(1).ok_or(Error::<T>::ArithmeticOverflow)?;
-        <TotalIssueCount>::set(count);
-
-        // TODO: fix this
-        let total = ext::treasury::get_total_supply::<T>();
-        let total_raw = Self::polkabtc_to_u128(total)?;
-        let average =
-            T::SignedFixedPoint::checked_from_rational(total_raw, count).ok_or(Error::<T>::TryIntoIntError)?;
+        let count = TotalIssueCount::mutate(|x| {
+            *x = x.saturating_add(1);
+            *x as u128
+        });
+        let total = LifetimeIssued::mutate(|x| {
+            *x = x.saturating_add(amount_raw);
+            *x
+        });
+
+        // calculate the average on raw input rather than in fixed_point - we don't want to fail
+        // if the result can not be losslessly represented. By using raw division we will be off
+        // but at most one Planck, which is acceptable
+        let average_raw = total.checked_div(count).ok_or(Error::<T>::ArithmeticOverflow)?;
```

Comment:
```suggestion
        let average_raw = total.checked_div(count).ok_or(Error::<T>::ArithmeticUnderflow)?;
```

Reply:
Technically it's neither underflow nor underflow. Of the two, I think overflow is more accurate, since division tends to infinity as `count` approaches 0